packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -29,7 +30,7 @@ var publicPath = '/';
 // `publicUrl` is just like `publicPath`, but we will provide it to our app
 // as %PUBLIC_URL% in `index.html` and `process.env.PUBLIC_URL` in JavaScript.
 // Omit trailing slash as %PUBLIC_PATH%/xyz looks better than %PUBLIC_PATH%xyz.
-var publicUrl = '';
+var publicUrl = ensureSlash(paths.servedPath, false);
```

Comment:
I expect that this might break HMR when `PUBLIC_URL` is specified. I think we might be assuming we're serving from the root somewhere either in `start.js` or in webpack hot client.